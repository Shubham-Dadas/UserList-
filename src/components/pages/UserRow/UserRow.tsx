import React, { Component } from "react";

interface UserProps {
  user: {
    id: number;
    name: string;
    email: string;
    gender: string;
    status: string;
  };
}

class UserRow extends Component<UserProps> {
  render() {
    const { name, email, gender, status } = this.props.user;

    return (
      <tr>
        <td>{name}</td>
        <td>{email}</td>
        <td>{gender}</td>
        <td>{status}</td>
      </tr>
    );
  }
}

export default UserRow;
