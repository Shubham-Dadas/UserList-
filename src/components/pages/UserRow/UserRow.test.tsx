import React from "react";
import UserRow from "./UserRow";
import { shallow } from "enzyme";

jest.mock("../../../services/service", () => ({
  getUsers: jest.fn(),
}));

 const mockUser = {
   id: 7705369,
   name: "Shubham Dadas",
   email: "shubham.dadas@15ce.com",
   gender: "male",
   status: "active",
 };


it("userRow snapshot matching", () => {
    const component = shallow(<UserRow user={mockUser} />)
    //@ts-ignore
    expect(component).toMatchSnapshot();
})
