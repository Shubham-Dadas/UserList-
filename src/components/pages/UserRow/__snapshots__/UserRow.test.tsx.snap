// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`userRow snapshot matching 1`] = `
<tr>
  <td>
    Shubham Dadas
  </td>
  <td>
    shubham.dadas@15ce.com
  </td>
  <td>
    male
  </td>
  <td>
    active
  </td>
</tr>
`;
